### DISCUSSION TITLE
Extend fi_domain_attr to include maximum buffer/VA range that can be registered

### DISCUSSION THREAD

**Discussion by swelch** [ofiwg/libfabric#7891] [General]
The attributes returned by fi_getinfo() do not report an attribute identifying the maximum VA range/or buffer length that the providers MR registration implementation can support. A client application should be able to determine the maximum size permitted directly from the providers fi_info attributes. This attribute is reported in other RDMA capable APIs, for instance via the IB Verbs ibv_query_device() call where it returns the maximum size in the returned ibv_device_attrs:

`uint64_t                max_mr_size;            /* Largest contiguous block that can be registered */`

Furthermore, it is possible that providers may support multiple address translation operation modes and this length may be different depending on a runtime selection.

We would like to propose that the struct fi_domain_attr be appended to include a similar attribute. For example:

```
struct fi_domain_attr {
        struct fid_domain.       *domain;
        char                               *name;
	.
	.
	.
        uint32_t                        tclass;
        size_t                            max_mr_size;
};
```

Any thoughts appreciated, thanks.

Steve

**Comment by shefty** [comment]
Moving to an open enhancement request.  Adding a single max_mr_size seems reasonable.  The current assumption is that it is the same as max_msg_size.  I really don't want to try exposing multiple values to the user based on other mr_attr values (e.g. remote vs local access, etc.)

**Reply by swelch** [reply]
Thanks for putting it in the correct spot. Agree, any single fi_info exported by the provider would have a single max_mr_size for the domain.
